### ISSUE TITLE
User record still in UserData table after uninstall

### ISSUE THREAD

**Issue by upcreek** [OfficeDev/microsoft-teams-apps-company-communicator#224]
Hello. Is the intended behaviour that when a user uninstalls the app from their personal scope, their record is deleted from the UserData table? This is not happening for me, users are still in UserData table and still receive messages from the app.

**Comment by DebGwynne** [comment]
We've got the same issue here, which has confused our testing as we add/remove people.  5 non-testers still being sent the notifications.  I'm off to strip them out of the usertable now manually to see what happens (feasible with 17 testers, not feasible if we had pushed it out to everyone!)

**Comment by upcreek** [comment]
Yes I think removing users manually from the table will stop them from receiving. Not ideal. I've also noticed that sometimes adding the receiving app to a Team will add their entry to the teams data table, but sometimes it won't. I'm thinking there is a transmission problem somewhere in Azure. Now trying to figure out how to debug it.

**Comment by mrogunlana** [comment]
> Yes I think removing users manually from the table will stop them from receiving. Not ideal. I've also noticed that sometimes adding the receiving app to a Team will add their entry to the teams data table, but sometimes it won't. I'm thinking there is a transmission problem somewhere in Azure. Now trying to figure out how to debug it.

@upcreek can you elaborate on your issue surrounding teams data being persisted intermittently (e.g. for example, what are the case(s) where you're seeing TeamsData being persisted)? I'm see the same results, just that I no longer see TeamsData in any case...

**Comment by DebGwynne** [comment]
@upcreek As you say - I manually removed the three unwanted entries from the user table and no messages went.  Lovely.  Until I messaged a team they were part of, and back into the user table they went.  Not sure if that's expected behaviour or not.

**Comment by upcreek** [comment]
@mrogunlana All my tests this week of adding the receiving app to a Team have been successful so far. Maybe it was a glitch? Testing ongoing....

@DebGwynne Now that you mention it, I've experienced similar behaviour. I've uninstalled the receiving app from my personal scope several times but every time I send a message to a Team that I belong to, the receiving app gets reinstalled in my personal scope. Very strange.
